allennlp/state_machines/states/lambda_grammar_statelet.py

```diff
@@ -0,0 +1,182 @@
+from typing import Callable, Dict, List, Tuple
+
+import torch
+
+from allennlp.nn import util
+
+
+# We're not actually inhereting from `GrammarStatelet` here because there's very little logic that
```

Comment:
This class is basically unmodified from the previous `GrammarStatelet`.  The only things that are different are the docstrings and this comment.